### ISSUE TITLE
False positive `invalid-enum-extension` for Enum parent class with annotations

### ISSUE THREAD

**Issue by rarredon** [pylint-dev/pylint#8830]
### Bug description

Pylint incorrectly flags an Enum inherited from a parent Enum which has annotations. This is useful when overriding `__new__` or `__init__` on the parent Enum to add extra data to the Enum members while also keeping the codebase typed. Example:

`example.py`
```
from enum import Enum


class ColorEnum(Enum):
    red: int
    green: int
    blue: int

    def __init__(self, red: int, green: int, blue: int) -> None:
        self.red = red
        self.green = green
        self.blue = blue

class Pastel(ColorEnum):
    SAGE = (170, 200, 167)
```

### Command used

```shell
pylint example.py
```

### Pylint output

```shell
************* Module example
example.py:14:0: E0244: Extending inherited Enum class "ColorEnum" (invalid-enum-extension)
```

### Expected behavior

No output

### Pylint version

```shell
pylint 2.17.4
astroid 2.15.5
Python 3.8.16 (default, Mar 15 2023, 13:36:46)
[Clang 14.0.0 (clang-1400.0.29.202)]
```

### OS / Environment

macOS Ventura 13.2.1

**Comment by Why-not-now** [comment]
I think you misunderstood how enum is used. For example, Enum is used to enumerate a list of limited selection of values (i.e. the days of the week). Here is an example of how Enum is used.
```py
from enum import Enum
class Weekday(Enum):
    MONDAY = 1
    TUESDAY = 2
    WEDNESDAY = 3
    THURSDAY = 4
    FRIDAY = 5
    SATURDAY = 6
    SUNDAY = 7
```
As you can see, Enum shouldn't be used to store values within its classes in the same way you store 3D coordinates into a class.

You can learn more about enum over [here](https://docs.python.org/3/howto/enum.html#enum-basic-tutorial).

**Comment by rarredon** [comment]
@Why-not-now there are several examples from the python docs involving Enums that store additional data to the Enum members. See for example https://docs.python.org/3/howto/enum.html#planet
